src/libs/ReportUtils.ts

```
@@ -1856,13 +1868,13 @@ function pushTransactionViolationsOnyxData(
         const isReportAnInvoice = isInvoiceReport(report);
 
         getReportTransactions(report.reportID).forEach((transaction: Transaction) => {
-            const transactionViolations = allTransactionViolations?.[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transaction.transactionID}`] ?? [];
+            const transactionViolations = allTransactionViolations?.[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transaction.transactionID}`];
```

why did we remove the default array here?

---

So, we set `value: transactionViolations ?? null,` in the `failureData` instead of an empty array.